### PR TITLE
TextEditor and HTML reader for the description field

### PR DESCRIPTION
**Pull request by nikomakela**
PT-547,
PT-1059 As a provider I can use an editor to style and create richer text,
PT-1060 As a provider I can see the editor styled text properly,
PT-1072 As a provider when I am previewing the event content from summary page I cannot get back to summary page.

### CODE REVIEW

File: src/domain/event/eventBasicInfo/EventBasicInfo.tsx [hunk 1]
```diff
@@ -45,6 +45,8 @@ const EventBasicInfo: React.FC<Props> = ({ eventData, language }) => {
       })
     : '';
 
+  const createMarkup = (markup: string) => ({ __html: markup });
```
Comment (Sartonon) [inline src/domain/event/eventBasicInfo/EventBasicInfo.tsx:48]: I think this could be just inlined

**Comment by nikomakela** [comment]
The textEditor test fails in github but not on my local machine.. 🤔

**Review by Sartonon — COMMENTED** [review]
Looking good :)

**Comment by Sartonon** [comment]
Screenshot 2021-07-01 at 6 36 05
Should background color of the editor be white? Also error message should be red?

File: src/common/components/textEditor/TextEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}) => {
+  const isMounted = useIsMounted();
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const editorRef = React.useRef<Editor>(null);
+  const [focused, setFocused] = React.useState(false);
+  const { t } = useTranslation();
+
+  const [editorState, setEditorState] = React.useState(
+    covertHtmlToEditorState(value)
+  );
```
[the remark this answers is not in the document]
Comment (nikomakela) [reply src/common/components/textEditor/TextEditor.tsx]: I think that the main idea is to convert it on every character 🤔

[the remark this answers is not in the document]
Comment (nikomakela) [reply src/common/components/textEditor/TextEditor.tsx]: at least I don't get what you mean. Maybe I'm just not familiar enough with the useCallback hook 😄

[the remark this answers is not in the document]
Comment (Sartonon) [reply src/common/components/textEditor/TextEditor.tsx]: it is useState hook

[the remark this answers is not in the document]
Comment (Sartonon) [reply src/common/components/textEditor/TextEditor.tsx]: Only first time the function return value has effect so subsequent render call are redundant. React.useState ignores future values so by just giving callback the function is called once.

[the remark this answers is not in the document]
Comment (nikomakela) [reply src/common/components/textEditor/TextEditor.tsx]: used your code and tested it - seems to work ok.

**Review by Sartonon — APPROVED** [review]
Looking good 😄

**Pull request merged**